libs/shared/src/main/java/io/crate/common/collections/Lists.java

```diff
@@ -104,6 +105,21 @@ public static <T> List<T> concatUnique(List<? extends T> list1, Collection<? ext
         return result;
     }
 
+    public static Collection<?> flatUnique(Collection<?> list) {
```

Comment:
minor, apologies I thought of it now: maybe `flattenUnique` as a verb sounds better, since it's a static method where you pass the list as arg, and not a member method like: `list.flatUnique()`.